.dotnet/src/Custom/OpenAIClient.cs

```diff
@@ -28,12 +28,16 @@ namespace OpenAI;
 [CodeGenSuppress("GetLegacyCompletionClientClient")]
 public partial class OpenAIClient
 {
+    protected readonly ApiKeyCredential _cachedCredential = null;
```

Comment:
If we make this visible in the public API, how are we thinking about using it when Azure clients use TokenCredential auth and not api key auth?

Reply:
Upon review: we don't really use this at all. The only place the credential is used is when creating the pipeline, which the PR centralizes. I'll remove this entirely and also expunge the handoffs as much as possible.